packages/reactive-element/src/decorators/base.ts

```diff
@@ -44,3 +46,30 @@ export const standardPrototypeMethod = (
   key: element.key,
   descriptor,
 });
+
+export const decorateProperty = (
```

Comment:
The idea is that this is a TS/babel normalization helper for writing your own property decorators, yeah? 

Can you add API docs for this and when it's intended to be used?

And a few questions: Do we intend end users to use this, or is it just for tests? Can you use this for defining all property decorators? Or just ones that are used in conjunction with `@property`? Is there a reason all of our property decorators couldn't/shouldn't be updated to use this helper for cleaner/smaller code?